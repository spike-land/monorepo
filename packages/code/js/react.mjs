import "es-module-shims"

export const 