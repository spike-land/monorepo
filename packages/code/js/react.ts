//import * as ReactM from "react";
//import * as ReactdoMCli from "react-dom/client";

const React = window.React = window.React || {};
const ReactDOM = window.ReactDOM = window.ReactDOM || {};

export const {
  Children,
  Component,
  Fragment,
  PureComponent,
  StrictMode,
  Suspense,
  cloneElement,
  createContext,
  createElement,
  createFactory,
  createRef,
  forwardRef,
  isValidElement,
  lazy,
  memo,
  startTransition,
  useCallback,
  useContext,
  useDebugValue,
  useDeferredValue,
  useEffect,
  useId,
  useImperativeHandle,
  useInsertionEffect,
  useLayoutEffect,
  useMemo,
  useReducer,
  useRef,
  useState,
  useSyncExternalStore,
  useTransition,
  version,
  __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED,
} = React;

export const {
  createRoot,
  flushSync,
  hydrateRoot,
  createPortal,
} = ReactDOM;

export default React;
