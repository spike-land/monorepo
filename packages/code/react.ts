import * as Preact from "preact";
import PreactCompat from "preact/compat";
import {
  hydrate as hy,
  render as rend,
  lazy as laz,
  Suspense as Sus, 
  SuspenseList as SusL,
  unmountComponentAtNode as unm,
} from "preact/compat";

import { renderToString as renderToStr } from "preact-render-to-string";

import {
  JSX as jjx,
  jsx as j,
  jsxDEV as jd,
  jsxs as js,
} from "preact/jsx-runtime";

window.ReactDOM = window.ReactDOM ||
  {
    hydrate: hy,
    render: rend,
    unmountComponentAtNode: unm,
    renderToString: renderToStr,
  };

window.PreactJSX = window.PreactJSX || { jsx: j, jsxDEV: jd, jsxs: js };
window.renderToString = window.renderToString || renderToStr;

window.React = window.React || {...PreactCompat,  lazy: laz,
  Suspense: Sus, 
  SuspenseList: SusL};

window.MyPreact = window.MyPreact || Preact;

export const { jsx, jsxDEV, jsxs } = window.PreactJSX;

export const { Fragment } = Preact;

export const { hydrate, render, unmountComponentAtNode } = window.ReactDOM;
// @ts-ignore
export const { renderToString } = window;

export const { toChildArray } = Preact;
export default PreactCompat;

export const {
  createContext,
  createElement,
  cloneElement,
  createFactory,
  createPortal,
  createRef,
  useCallback,
  useContext,
  useDebugValue,
  isValidElement,
  useEffect,
  useImperativeHandle,
  useLayoutEffect,
  useMemo,
  useReducer,
  useRef,
  useState,
  lazy,
  Suspense,
  StrictMode,
  SuspenseList,
  findDOMNode,
  forwardRef,
  memo,
  Children,
  PureComponent,
  Component,
  version,
} = window.React;

export const flushSync = (callback: (arg: boolean) => void, arg: boolean) =>
  callback(arg);

export function createRoot(container: HTMLDivElement) {
  return {
    render(children: preact.VNode<any>) {
      render(children, container);
    },
    unmount() {
      unmountComponentAtNode(container);
    },
  };
}

export function hydrateRoot(
  container: HTMLDivElement,
  children: preact.VNode<any>,
) {
  hydrate(children, container);
  return createRoot(container);
}
