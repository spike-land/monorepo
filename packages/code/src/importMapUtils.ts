export const oo: { [key: string]: string } = {
  "@emotion/react/jsx-runtime": "/emotionJsxRuntime.mjs",
  "react/jsx-runtime": "/jsx.mjs",
  "react-dom/server": "/reactDomServer.mjs",
  "react-dom/client": "/reactDomClient.mjs",
  "@emotion/react": "/emotion.mjs",
  "react": "/reactMod.mjs",
  "framer-motion": "/motion.mjs",
  "react-dom": "/reactDom.mjs",
  "foo-bar": "/fooBar.mjs",
  "recharts": "/recharts.mjs",
};

export const importMap = { imports: oo };

export function importMapReplace(code: string, origin: string): string {
  // Check if the code has already been processed
  if (code.startsWith("/** importMapReplace */")) {
    return code;
  }

  const importRegex = /^([ \t]*import\s+(?:[\w*{}\n\r\t, ]+)\s+from\s+)["'](.*[@\w_-]+)["']\s*;?\s*(\/\/.*)?\s*$/gm;
  const dynamicImportRegex = /import\((["'`])(.+?)\1\)/g;
  const exportRegex = /^([ \t]*export\s+(?:{\s*[\w\s,]+\s*}\s+from\s+)?)["'](.*[@\w_-]+)["']\s*;?\s*(\/\/.*)?\s*$/gm;

  function replaceImport(match: string, importStatement: string, path: string, inlineComment: string = ""): string {
    let newPath = path;

    if (oo[path]) {
      newPath = `${origin}${oo[path]}`;
    } else if (!path.startsWith("http") && !path.startsWith("./") && !path.startsWith("../")) {
      if (path.startsWith("@/")) {
        newPath = `${origin}/@/${path.slice(2)}`;
      } else {
        newPath = `${origin}/${path}`;
      }
      if (!newPath.includes("?bundle")) {
        newPath += path === "axios" ? "?bundle" : ".mjs";
      }
    } else if (path.startsWith("./") || path.startsWith("../")) {
      newPath = `${origin}/live/${path.replace(/^\.\//, "")}`;
      if (!newPath.endsWith(".js") && !newPath.endsWith(".mjs")) {
        newPath += "/index.js";
      }
    } else {
      return match;
    }

    // Extract specific exports if present
    const exportsMatch = importStatement.match(
      /{\s*([\w\s,]+(?:\s+as\s+[\w]+)?(?:\s*,\s*[\w]+(?:\s+as\s+[\w]+)?)*)\s*}/,
    );
    if (exportsMatch && !oo[path] && !path.startsWith("@/")) {
      const specificExports = exportsMatch[1].replace(/\s+as\s+/g, " as ").trim();
      newPath += newPath.includes("?") ? "&" : "?";
      newPath += `exports=${encodeURIComponent(specificExports)}`;
    }

    return `${importStatement}"${newPath}";${inlineComment ? " " + inlineComment : ""}`;
  }

  function replaceDynamicImport(match: string, quote: string, path: string): string {
    if (path.includes("${")) {
      return match;
    }
    if (oo[path]) {
      return `import(${quote}${origin}${oo[path]}${quote})`;
    }
    if (!path.startsWith("http") && !path.startsWith("./") && !path.startsWith("../")) {
      if (path.startsWith("@/")) {
        return `import(${quote}${origin}/@/${path.slice(2)}?bundle${quote})`;
      }
      return `import(${quote}${origin}/${path.includes("*") ? path : "*" + path + "?bundle"}${quote})`;
    }
    return match;
  }

  function replaceExport(match: string, exportStatement: string, path: string, inlineComment: string = ""): string {
    let newPath = path;

    if (oo[path]) {
      newPath = `${origin}${oo[path]}`;
    } else if (!path.startsWith("http") && !path.startsWith("./") && !path.startsWith("../")) {
      if (path.startsWith("@/")) {
        newPath = `${origin}/@/${path.slice(2)}`;
      } else {
        newPath = `${origin}/${path}`;
      }
      newPath += ".mjs";
    } else if (path.startsWith("./") || path.startsWith("../")) {
      newPath = `${origin}/live/${path.replace(/^\.\//, "")}`;
      if (!newPath.endsWith(".js") && !newPath.endsWith(".mjs")) {
        newPath += "/index.js";
      }
    } else {
      return match;
    }

    return `${exportStatement}"${newPath}";${inlineComment ? " " + inlineComment : ""}`;
  }

  code = code.replace(importRegex, replaceImport);
  code = code.replace(dynamicImportRegex, replaceDynamicImport);
  code = code.replace(exportRegex, replaceExport);

  return `/** importMapReplace */\n${code.trim()}`;
}
