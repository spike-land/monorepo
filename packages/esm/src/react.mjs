const React = window.React;

export const { createContext } = React;
export const { useDebugValue } = React;
export const { useState } = React;
export const { useId } = React;
export const { useRef } = React;
export const { useContext } = React;

export const { useEffect } = React;
export const { useLayoutEffect } = React;
export const { useReducer } = React;
export const { useCallback } = React;
export const { forwardRef } = React;
export const { createElement } = React;
export const { createFactory } = React;
export const { createRef } = React;
export const { Fragment } = React;
export const { Component } = React;

export const { isValidElement } = React;
export const { memo } = React;
export const { useImperativeHandle } = React;

export const { Children } = React;
export const { cloneElement } = React;

export const { useMemo } = React;

export default React;
