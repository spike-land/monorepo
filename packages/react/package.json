{
  "name": "@zedvision/emotion-react",
  "version": "10.10.16",
  "main": "dist/emotion-react.cjs.js",
  "module": "dist/emotion-react.esm.js",
  "browser": {
    "./dist/emotion-react.cjs.js": "./dist/emotion-react.browser.cjs.js",
    "./dist/emotion-react.esm.js": "./dist/emotion-react.browser.esm.js"
  },
  "types": "types/index.d.ts",
  "files": [
    "src",
    "dist",
    "jsx-runtime",
    "jsx-dev-runtime",
    "isolated-hoist-non-react-statics-do-not-use-this-in-your-code",
    "types/*.d.ts",
    "macro.js",
    "macro.d.ts",
    "macro.js.flow"
  ],
  "sideEffects": false,
  "author": "mitchellhamilton <mitchell@mitchellhamilton.me>",
  "license": "MIT",
  "scripts": {
    "test:typescript": "dtslint types",
    "build": "microbundle build"
  },
  "dependencies": {
    "@babel/runtime": "^7.12.5",
    "@zedvision/emotion-cache": "^10.10.16",
    "@zedvision/emotion-serialize": "^10.10.16",
    "@zedvision/emotion-sheet": "^10.10.16",
    "@zedvision/emotion-utils": "^10.10.16",
    "@zedvision/emotion-weak-memoize": "^10.10.16",
    "esbuild": "^0.8.26",
    "hoist-non-react-statics": "^3.3.2",
    "typescript": "^4.1.3"
  },
  "peerDependencies": {
    "@babel/core": "^7.12.10",
    "react": ">=17.0.1"
  },
  "peerDependenciesMeta": {
    "@babel/core": {
      "optional": true
    },
    "@types/react": {
      "optional": true
    }
  },
  "workspaces": [
    "packages/*"
  ],
  "private": "false",
  "devDependencies": {
    "@babel/core": "^7.12.10",
    "@types/react": "^17.0.0",
    "@zedvision/emotion-css": "^10.10.15",
    "@zedvision/emotion-css-prettifier": "^10.10.16",
    "@zedvision/emotion-server": "^0.0.5",
    "@zedvision/emotion-styled": "^0.0.5",
    "dtslint": "^4.0.6",
    "html-tag-names": "^1.1.5",
    "react": "17.0.1",
    "svg-tag-names": "^2.0.1"
  },
  "repository": "https://github.com/@zedvision/emotion-js/emotion/tree/master/packages/react",
  "publishConfig": {
    "access": "public"
  },
  "umd:main": "dist/emotion-react.umd.min.js",
  "preconstruct": {
    "entrypoints": [
      "./index.js",
      "./jsx-runtime.js",
      "./jsx-dev-runtime.js",
      "./isolated-hoist-non-react-statics-do-not-use-this-in-your-code.js"
    ],
    "umdName": "emotionReact"
  },
  "gitHead": "ae10058048862c5ae2744259d3cfca958f01ff98"
}
