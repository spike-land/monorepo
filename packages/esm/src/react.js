import * as React from "react";

export const mods = {...React}

export default React
