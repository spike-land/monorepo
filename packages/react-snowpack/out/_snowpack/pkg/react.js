import { r as react } from './common/index-f537190e.js';
export { r as default } from './common/index-f537190e.js';
import './common/_commonjsHelpers-eb5a497e.js';
