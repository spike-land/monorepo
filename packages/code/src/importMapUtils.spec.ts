import { afterAll, beforeAll, describe, expect, it, vi } from "vitest";
import { importMapReplace } from "./importMapUtils";

describe("importMapReplace", () => {
  const origin = "http://localhost:3000";

  beforeAll(() => {
    vi.stubGlobal(
      "fetch",
      vi.fn().mockImplementation(() =>
        Promise.resolve({
          ok: true,
          json: () =>
            Promise.resolve({
              "name": "some-module",
              "version": "1.0.0",
              "main": "index.js",
              "browser": "browser.js",
              "module": "module.js",
              "types": "index.d.ts",
              "files": [
                "index.js",
                "browser.js",
                "module.js",
                "index.d.ts",
              ],
            }),
        })
      ),
    );
  });

  afterAll(() => {
    vi.unstubAllGlobals();
  });

  it("should replace top-level imports", () => {
    const code = 'import React from "react";';
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import React from "${origin}/reactMod.mjs"`);
  });

  it("should replace multiple top-level imports", () => {
    const code = `
      import React from "react";
      import { useState, useEffect } from "react";
    `;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import React from "${origin}/reactMod.mjs"`);
    expect(result).toContain(`import { useState, useEffect } from "${origin}/reactMod.mjs"`);
  });

  it("should replace imports with aliases", () => {
    const code = `import { useState as useStateAlias } from "react";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import { useState as useStateAlias } from "${origin}/reactMod.mjs"`);
  });

  it("should replace dynamic imports", () => {
    const code = `const React = import("react");`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`const React = import("${origin}/reactMod.mjs"`);
  });

  it("should not replace dynamic imports with template literals", () => {
    const code = "const dynamic = 'react'; const React = import(`${dynamic}`);";
    const result = importMapReplace(code, origin);
    expect(result).toContain(`const React = import(\`\${dynamic}\`)`);
  });

  it("should replace imports from @/ paths", () => {
    const code = `import { Button } from "@/components/ui/button";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import { Button } from "${origin}/@/components/ui/button.mjs"`);
  });

  it("should replace relative imports", () => {
    const code = `import MyComponent from "./MyComponent";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import MyComponent from "${origin}/live/MyComponent/index.js"`);
  });

  it("should not replace absolute URLs", () => {
    const code = `import MyComponent from "https://example.com/MyComponent.js";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`/** importMapReplace */\n${code}`);
  });

  it("should replace exports", () => {
    const code = `export { default } from "react";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`export { default } from "${origin}/reactMod.mjs"`);
  });

  it("should handle multiple imports and exports", () => {
    const code = `
      import React from "react";
      import { useState } from "react";
      export { Button } from "@/components/ui/button";
      const lodash = import("lodash");
    `;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import React from "${origin}/reactMod.mjs"`);
    expect(result).toContain(`import { useState } from "${origin}/reactMod.mjs"`);
    expect(result).toContain(`export { Button } from "${origin}/@/components/ui/button.mjs"`);
    expect(result).toContain(`const lodash = import("${origin}/*lodash?bundle")`);
  });

  it("should handle imports with comments", () => {
    const code = `
      // This is a comment
      import React from "react"; // Inline comment
      /* Multi-line
         comment */
      import { useState } from "react";
    `;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import React from "${origin}/reactMod.mjs"`);
    expect(result).toContain(`import { useState } from "${origin}/reactMod.mjs"`);
  });

  it("should handle complex scenarios", () => {
    const code = `
      import React, { useState, useEffect as useEffectAlias } from "react";
      import { Button } from "@/components/ui/button";
      const lodash = import("lodash");
      export { default as MyComponent } from "./MyComponent";
      const dynamicImport = (module) => import(\`\${module}\`);
    `;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import React, { useState, useEffect as useEffectAlias } from "${origin}/reactMod.mjs"`);
    expect(result).toContain(`import { Button } from "${origin}/@/components/ui/button.mjs"`);
    expect(result).toContain(`const lodash = import("${origin}/*lodash?bundle")`);
    expect(result).toContain(`export { default as MyComponent } from "${origin}/live/MyComponent/index.js"`);
    expect(result).toContain(`const dynamicImport = (module) => import(\`\${module}\`)`);
  });

  it("should add importMapReplace comment", () => {
    const code = `import React from "react";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`/** importMapReplace */`);
  });

  it("should handle imports with file extensions", () => {
    const code = `
      import MyModule from "./myModule.js";
      import MyTypeScriptModule from "./myModule.ts";
      import MyJSXModule from "./myModule.jsx";
      import MyTSXModule from "./myModule.tsx";
    `;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import MyModule from "${origin}/live/myModule.js/index.js"`);
    expect(result).toContain(`import MyTypeScriptModule from "${origin}/live/myModule.ts/index.js"`);
    expect(result).toContain(`import MyJSXModule from "${origin}/live/myModule.jsx/index.js"`);
    expect(result).toContain(`import MyTSXModule from "${origin}/live/myModule.tsx/index.js"`);
  });

  it("should handle imports with query parameters", () => {
    const code = `
      import MyModule from "my-module?param1=value1&param2=value2";
    `;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import MyModule from "${origin}/*my-module?param1=value1&param2=value2?bundle"`);
  });

  it("should handle react-spring imports", () => {
    const code = `
      import { animated, useSpring } from "react-spring";
    `;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import { animated, useSpring } from "${origin}/*react-spring?bundle&exports=animated%2C%20useSpring"`);
  });

  it("should not modify code that already contains importMapReplace", () => {
    const code = `
/** importMapReplace */
      import React from "http://localhost:3000/reactMod.mjs";
    `;
    const result = importMapReplace(code, origin);
    expect(result).toBe(code);
  });

  it("should handle specific named exports", () => {
    const code = `import { prop, prop2 } from "foo";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import { prop, prop2 } from "${origin}/*foo?bundle&exports=prop%2C%20prop2"`);
  });

  it("should handle specific exports from tslib", () => {
    const code = `import { __await, __rest } from "tslib";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import { __await, __rest } from "${origin}/*tslib?bundle&exports=__await%2C%20__rest"`);
  });

  it("should handle specific named imports with aliases", () => {
    const code = `import { __await as aw, __rest as restNow } from "http://localhost:3000/*tslib?bundle&exports=__await%20as%20aw%2C%20__rest%20as%20restNow";`;
    const result = importMapReplace(code, origin);
    expect(result).toContain(`import { __await as aw, __rest as restNow } from "${origin}/*tslib?bundle&exports=__await%2C%20__rest"`);
  });
});
