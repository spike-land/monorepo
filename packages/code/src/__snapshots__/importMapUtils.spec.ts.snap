// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`importMapReplace > should apply custom mappings 1`] = `
"
/** importMapReplace */
import CustomPackage from "/custom-package?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true";"
`;

exports[`importMapReplace > should ignore absolute URLs 1`] = `
"
/** importMapReplace */
import MyComponent from "http://example.com/MyComponent.js";"
`;

exports[`importMapReplace > should ignore relative URLs 1`] = `
"
/** importMapReplace */
import MyComponent from "/live/MyComponent/index.js";"
`;

exports[`importMapReplace > should not do anything 1`] = `
"
/** importMapReplace */
 eturn e[t] === r ||
              (e[t] = r,
                (t === "prop" || t === "value" || t === "name" ||
                  t === "params" || t === "important" || t === "text") &&
                e.markDirty()),
              !0;"
`;

exports[`importMapReplace > should process this 1`] = `
"
/** importMapReplace */
eturn d.className=f,a&&(d.ref=a),O.createElement(O.Fragment,null,O.createElement(On,{cache:t,serialized:m,isStringTag:typeof c=="string"}),O.createElement(c,d))}),dr=Rn;import"react";var va=Y(Yr());var xa=Z.Fragment;function Sa(e,t,a){return me.call(t,"css")?Z.jsx(dr,pr(e,t),a):Z.jsx(e,t,a)}function Ea(e,t,a){return me.call(t,"css")?Z.jsxs(dr,pr(e,t),a):Z.jsxs(e,t,a)}";"
`;

exports[`importMapReplace > should replace DYNAMID shadcdn imports 1`] = `
"
/** importMapReplace */
import React from "/@/components/ui/alert.mjs";"
`;

exports[`importMapReplace > should replace coder44  1`] = `
"
/** importMapReplace */

    import { css } from "@emotion/react";
    import $ from "react";

    import { css  as $kdkd } from "@emotion/react";

    import { useEffect, useState, useRef } from "react";
    import type { FC } from "react";
    import QRTerminal from "/qrcode-terminal?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true";
    "
`;

exports[`importMapReplace > should replace dynamic imports 1`] = `
"
/** importMapReplace */
const React = import("react");"
`;

exports[`importMapReplace > should replace dynamic imports 2`] = `
"
/** importMapReplace */
const React = import("react");"
`;

exports[`importMapReplace > should replace dynamic imports 3`] = `
"
/** importMapReplace */
const module = import("/some-module?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true");"
`;

exports[`importMapReplace > should replace dynamic imports at the end 1`] = `
"
/** importMapReplace */

    import{Buffer as De}from"/v125/buffer@6.0.3/es2024/buffer.bundle.mjs"
    let { Mp3Encoder: t } = await import("/lamejs?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true");
    import{timeDay as _,timeSunday as bt,timeMonday as b,timeThursday as w,timeYear as S,utcDay as $,utcSunday as Vt,utcMonday as V,utcThursday as p,utcYear as F}from"/d3-time?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true&exports=timeDay,timeSunday,timeMonday,timeThursday,timeYear,utcDay,utcSunday,utcMonday,utcThursday,utcYear";
      async function loadModule() {
        let{default:c}=await import("/axios?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true");
        let{default:d}=await import("https://unpkg.com/axios/dist/axios.min.js");
        
        const module = await import("/some-module?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true");
        // do something with module
      }
        import"react/jsx-runtime";
        import"react";
        async function y(s){await s(m)}
        export{n as Particles,n as default,y as initParticlesEngine};
    "
`;

exports[`importMapReplace > should replace dynamic imports at the end 2`] = `
"
/** importMapReplace */

    import { useSpring, animated } from "/react-spring?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true&exports=useSpring,animated"; // Importing react-spring
    "
`;

exports[`importMapReplace > should replace dynamic imports inside a function 1`] = `
"
/** importMapReplace */

    import{Buffer as De}from"/v125/buffer@6.0.3/es2024/buffer.bundle.mjs"
    let { Mp3Encoder: t } = await import("/lamejs?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true");
    import{timeDay as _,timeSunday as bt,timeMonday as b,timeThursday as w,timeYear as S,utcDay as $,utcSunday as Vt,utcMonday as V,utcThursday as p,utcYear as F}from"/d3-time?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true&exports=timeDay,timeSunday,timeMonday,timeThursday,timeYear,utcDay,utcSunday,utcMonday,utcThursday,utcYear";
      async function loadModule() {
        let{default:c}=await import("/axios?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true");
        let{default:d}=await import("https://unpkg.com/axios/dist/axios.min.js");
        
        const module = await import("/some-module?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true");
        // do something with module
      }
    "
`;

exports[`importMapReplace > should replace dynamic imports with await 1`] = `
"
/** importMapReplace */
const module = await import("/some-module?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true");"
`;

exports[`importMapReplace > should replace dynamic imports2 1`] = `
"
/** importMapReplace */
const dynamic = 'dd'; const React = import(\`\${dynamic}dom\`);"
`;

exports[`importMapReplace > should replace shadcdn  1`] = `
"
/** importMapReplace */

    import { css } from "@emotion/react";
    import $ from "react";
    import { Button } from "/@/components/ui/button.mjs";

    import { css  as $kdkd } from "@emotion/react";

    import { useEffect, useState, useRef } from "react";
    import type { FC } from "react";
    import QRTerminal from "/qrcode-terminal?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true";
    "
`;

exports[`importMapReplace > should replace the live urls 1`] = `
"
/** importMapReplace */
import Box from "/live/Box/index.js";"
`;

exports[`importMapReplace > should replace top-level exports 1`] = `
"
/** importMapReplace */
export { default } from "react";"
`;

exports[`importMapReplace > should replace top-level exports, except the ones in importmap 1`] = `
"
/** importMapReplace */
import { useState } from "react";"
`;

exports[`importMapReplace > should replace top-level imports 1`] = `
"
/** importMapReplace */
import React from "react";"
`;

exports[`importMapReplace > should replace top-level imports with dot in the path 1`] = `
"
/** importMapReplace */
import React from "/react.gl?external=@emotion/react/jsx-runtime,react/jsx-runtime,react-dom/server,react-dom/client,@emotion/react,react,framer-motion,react-dom,recharts&bundle=true";"
`;

exports[`importMapReplace > should replace top-leveldd imports with dot in the path 1`] = `
"
/** importMapReplace */
import React from "/live/box/index.js";"
`;
